### PR TITLE
Remove the need for UAC

### PR DESCRIPTION
**Pull request by ThePixelGamer**
This should remove the requirement for admin permissions, not super well tested but as far as I can tell downloading/registering Releases and Betas seem to work fine.

### CODE REVIEW

**Comment by ThePixelGamer** [comment]
Writing this before this is considered but seems to break registering from a second user account, catch the error and elevate?

**Comment by MCMrARM** [comment]
Yeah a proper PR to remove UAC would handle auto-elevation in the case it's needed for package removal.

**Comment by MCMrARM** [comment]
The way I'd probably do it is have the main executable have an argument like /Uninstall and just call the main program with it.

**Comment by ThePixelGamer** [comment]
> Something I previously considered for this was to have a helper executable which would make a UAC prompt only when elevation is required, and only deal with removing the packages.

Doesn't seem to be possible without some COM code so for this PR it seems like mrarm's suggestion would be the way to go

**Comment by dktapps** [comment]
It should just be a case of invoking a subprocess with arguments, no?

**Comment by ThePixelGamer** [comment]
Sorry I misread the comment, that's actually what I'm working on right

**Comment by ThePixelGamer** [comment]
This is turning into a bigger PR than I intended as it seems multiple users support is pretty limited? Are y'all able to test installing with this launcher using 2 separate admin user accounts? From my testing it doesn't really work but it could be the result of a change I've made

**Comment by dktapps** [comment]
I've tested it and it works to the best of my knowledge.

**Comment by ThePixelGamer** [comment]
> I've tested it and it works to the best of my knowledge.

Does you setup have a wpsystem folder on another drive? For some reason my minecraft data is being "junctioned" to this folder

**Comment by dktapps** [comment]
Not me. Only standard C: setup for me.

**Comment by ThePixelGamer** [comment]
This commit removes the app package if it's been installed by another user which is required for development packages (2 users cannot have the same one installed). 
More changes will be needed for this to be ready but I wanted to get y'all's opinion on the state of it.
Testing: There's a hardcoded version string at line 243 of MainWindow.xaml.cs, make sure it matches whatever version you install with the 2nd user.

File: MCLauncher/MainWindow.xaml.cs [hunk 1]
```diff
@@ -213,15 +214,41 @@ public partial class MainWindow : Window, ICommonVersionCommands {
...
+                        startInfo.WorkingDirectory = Environment.CurrentDirectory;
+                        startInfo.FileName = Assembly.GetExecutingAssembly().Location;
+                        startInfo.Verb = "runas";
+                        startInfo.Arguments = "/Uninstall";
+                        try {
+                            Process elevated_process = Process.Start(startInfo);
+                            elevated_process.WaitForExit();
+                        }
+                        catch (System.ComponentModel.Win32Exception ex) {
+                            src.SetException(new Exception("Failed to unintall using elevated app: " + ex.Message));
```
Comment (dktapps) [inline MCLauncher/MainWindow.xaml.cs:238]:
```suggestion
                            src.SetException(new Exception("Failed to uninstall using elevated app: " + ex.Message));
```

File: MCLauncher/MainWindow.xaml.cs [hunk 2]
```diff
@@ -236,8 +263,18 @@ public partial class MainWindow : Window, ICommonVersionCommands {
...
-        private void BackupMinecraftDataForRemoval(string packageFamily) {
+        private string GetMinecraftDataFolder(string packageFamily) {
             var data = ApplicationDataManager.CreateForPackageFamily(packageFamily);
+            return data.LocalFolder.Path;
+        }
+
+        private void BackupMinecraftDataForRemoval(string packageFamily) {
+            string dir = GetMinecraftDataFolder(packageFamily);
+            if (!Directory.Exists(dir)) {
+                Debug.WriteLine("BackupMinecraftDataForRemoval error: " + packageFamily + " doesn't exists");
```
Comment (dktapps) [inline MCLauncher/MainWindow.xaml.cs:274]:
```suggestion
                Debug.WriteLine("BackupMinecraftDataForRemoval error: " + packageFamily + " doesn't exist");
```

File: MCLauncher/MainWindow.xaml.cs [hunk 3]
```diff
@@ -213,15 +214,41 @@ public partial class MainWindow : Window, ICommonVersionCommands {
...
+                            Process elevated_process = Process.Start(startInfo);
+                            elevated_process.WaitForExit();
+                        }
+                        catch (System.ComponentModel.Win32Exception ex) {
+                            src.SetException(new Exception("Failed to unintall using elevated app: " + ex.Message));
+                            return;
+                        }
+
+                        Debug.WriteLine("Rerunning installer");
+                        string gameDir = Path.GetFullPath("Minecraft-1.14.60.5");
```
[the remark this answers is not in the document]
Comment (ThePixelGamer) [reply MCLauncher/MainWindow.xaml.cs:243]: I wasn't sure how I'd get the version state to this part of the code, any suggestions on how?

**Comment by ThePixelGamer** [comment]
That should be good functionally but I don't know if I like it coding wise

**Pull request closed without merging**